src/test/txvalidationcache_tests.cpp

```diff
@@ -184,10 +184,10 @@ BOOST_FIXTURE_TEST_CASE(checkinputs_test, TestChain100Setup)
     // Sign, with a non-DER signature
     {
         std::vector<unsigned char> vchSig;
-        uint256 hash = SignatureHash(p2pk_scriptPubKey, spend_tx, 0, SIGHASH_ALL, 0, SIGVERSION_BASE);
+        uint256 hash = SignatureHash(p2pk_scriptPubKey, spend_tx, 0, SIGHASH_ALL|SIGHASH_FORKID, 0, SIGVERSION_BASE);
         BOOST_CHECK(coinbaseKey.Sign(hash, vchSig));
         vchSig.push_back((unsigned char) 0); // padding byte makes this non-DER
-        vchSig.push_back((unsigned char)SIGHASH_ALL);
+        vchSig.push_back((unsigned char)(SIGHASH_ALL|SIGHASH_FORKID));
```

Comment:
I'm wondering: given that it's always mandatory, why not update the value for the old sighash flags instead of combining both all the time?

Reply:
It's also just a maintainability nightmare to combine flags and have them mean different things on different implementations from what they are understood and documented to mean.